### PR TITLE
move routes generation to a routes helper file

### PR DESCRIPTION
**Pull request by gregmolnar**

### CODE REVIEW

**Comment by excid3** [comment]
For test coverage, we could probably just start with some simple integration tests.

**Comment by excid3** [comment]
For unit tests, looks like we can create a new route set and add routes to it:

https://github.com/heartcombo/devise/blob/d0f0853c75defb10d83f4b2576b8df9e4cbb3695/test/routes_test.rb#L262-L264

This is pretty cool!

**Comment by gregmolnar** [comment]
I added a simple integration test for the sessions controller only, maybe someone would want to add coverage for the rest of the controllers based on that if we open an issue for it? If not, I will do it later. 
I will also look into those unit tests at some point. 
There is an unrelated commit to allow a blank password on update, otherwise if you want to update the model without providing the password it will be invalid.

File: lib/revise_auth/model.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ module Model
 
       validates :email, format: {with: URI::MailTo::EMAIL_REGEXP}, presence: true, uniqueness: true
       validates :unconfirmed_email, format: {with: URI::MailTo::EMAIL_REGEXP}, allow_blank: true
-      validates_length_of :password, minimum: 12
+      validates_length_of :password, minimum: 12, allow_nil: true
```
Comment (excid3) [inline lib/revise_auth/model.rb:11]: What's the reason to allow nil?

Comment (gregmolnar) [reply lib/revise_auth/model.rb:11]: To make it possible to update the user model without providing the password. It is still required on create even with this.

Comment (excid3) [reply lib/revise_auth/model.rb:11]: Ah right

**Pull request merged**